### DISCUSSION TITLE
I want to customize syntax coloring of code rendered in views of my repository

### DISCUSSION THREAD

**Discussion by llaenowyd** [community/community#22067] [New to GitHub]
<p>I see github applies a syntax coloring to code in view, it seems to be completely static, different rules for different languages but not extensible or customizable.</p>
<p>I wish there was a way to set this up by the repository owner, not a user setting in the browser for reinterpreting github’s rendering.</p>
<p>Same as for whitespace decisions that are built into the code and displayed on github as-is, but it has to be out of band from the code in the case of the syntax coloring.</p>
<p>Use case is rainbow brackets, they help a lot with reading functional code.  Looking at my code in my repository, although I tried to make it readable when I was writing it in the IDE where I have a rainbow brackets plugin, it is not as readable on github where it’s presented for others to try to read.</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi <a class="mention" href="/llaenowyd">@llaenowyd</a>! :wave: Welcome to the Community!</p>
<p>That would be a neat feature, yes! Syntax highlighting is not actually set by GitHub itself, though.</p>
<p>We use open source TextMate-style language grammars for syntax highlighting, which are available here:</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/github/linguist/blob/96ad1185828f44bb9b774328a584551ee57ed264/grammars.yml" target="_blank" rel="noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/github/linguist/blob/96ad1185828f44bb9b774328a584551ee57ed264/grammars.yml" target="_blank" rel="noopener">github/linguist/blob/96ad1185828f44bb9b774328a584551ee57ed264/grammars.yml</a></h4>
<pre><code class="lang-yml">https://bitbucket.org/Clams/sublimesystemverilog/get/default.tar.gz:
- source.systemverilog
- source.ucfconstraints
https://svn.edgewall.org/repos/genshi/contrib/textmate/Genshi.tmbundle/Syntaxes/Markup%20Template%20%28XML%29.tmLanguage:
- text.xml.genshi
vendor/grammars/Alloy.tmbundle:
- source.alloy
vendor/grammars/Assembly-Syntax-Definition:
- source.x86
vendor/grammars/Atom-PostScript:
- source.postscript
vendor/grammars/AutoHotkey:
- source.ahk
vendor/grammars/BrightScript.tmbundle:
- source.brightauthorproject
- source.brightscript
vendor/grammars/ColdFusion:
- source.cfscript
- source.cfscript.cfc
- text.cfml.basic
</code></pre>

  This file has been truncated. <a href="https://github.com/github/linguist/blob/96ad1185828f44bb9b774328a584551ee57ed264/grammars.yml" target="_blank" rel="noopener">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>You could find the issue tracker for the specific grammar you’re interested in and open an issue (or contribute a feature!) there. Linguist pulls in grammar updates with each new release, which usually happens every couple of weeks. This should get fixed on <a href="http://GitHub.com">GitHub.com</a> shortly after that happens!</p>
<p>I hope this is helpful!</p>

**Reply by RedCMD** [reply]
but rainbow brackets are not controlled by TextMate grammars?
they would need to be implemented by GitHub itself
Like how VSCode does it
